tensorflow_lite_support/cc/test/task/vision/image_transformer_test.cc

```diff
@@ -0,0 +1,145 @@
+
+  std::unique_ptr<FrameBuffer> frame_buffer = CreateFromRgbRawBuffer(
+      rgb_image.pixel_data,
+      FrameBuffer::Dimension{rgb_image.width, rgb_image.height});
+  ImageTransformerOptions options;
+  options.mutable_base_options()->mutable_model_file()->set_file_name(
+    JoinPath("./" /*test src dir*/, kTestDataDirectory,
+              kESRGANModelWithInputMetaData));
+  
+  SUPPORT_ASSERT_OK_AND_ASSIGN(std::unique_ptr<ImageTransformer> image_transformer,
+                      ImageTransformer::CreateFromOptions(options));
+
+  StatusOr<FrameBuffer> result_or =
+      image_transformer->Transform(*frame_buffer);
+  ImageDataFree(&rgb_image);
+  SUPPORT_ASSERT_OK(result_or);
+}
+
+class SuperResolutionTest : public tflite_shims::testing::Test {};
+
+// Calculate the peak signal-to-noise ratio.
+// Original code: https://www.geeksforgeeks.org/python-peak-signal-to-noise-ratio-psnr/.
+double PSNR(const FrameBuffer& enhancedImage, const FrameBuffer& testImage) {
+  int imageSize = testImage.dimension().width * testImage.dimension().height;
+  const uint8* enhancedImagePtr = enhancedImage.plane(0).buffer;
+  const uint8* testImagePtr = testImage.plane(0).buffer;
+  double mse = 0.0;
+  for (int i = 0; i < imageSize; ++i, ++enhancedImagePtr, ++testImagePtr) {
+      mse += std::pow(static_cast<double>(*enhancedImagePtr) - static_cast<double>(*testImagePtr), 2);
+  }
+  mse /= imageSize;
+
+  // Zero MSE means no noise is present in the signal.
+  double psnr = mse == 0 ? 100.0 : 20 * std::log10(255.0 / std::sqrt(mse));
+
+  return psnr;
+}
+
+// Use a bi-cubically downsampled  image as input to the model and compare
+// the model output with the original image.
+TEST_F(SuperResolutionTest, GoldenImageComparisonTest) {
+  SUPPORT_ASSERT_OK_AND_ASSIGN(ImageData husky_downsampled, LoadImage("husky_downsampled.jpg"));
+  SUPPORT_ASSERT_OK_AND_ASSIGN(ImageData husky_original, LoadImage("husky_original.jpg"));
+
+  std::unique_ptr<FrameBuffer> husky_downsampled_buffer = CreateFromRgbRawBuffer(
+      husky_downsampled.pixel_data,
+      FrameBuffer::Dimension{husky_downsampled.width, husky_downsampled.height});
+
+  std::unique_ptr<FrameBuffer> husky_original_buffer = CreateFromRgbRawBuffer(
+      husky_original.pixel_data,
+      FrameBuffer::Dimension{husky_original.width, husky_original.height});
+
+  ImageTransformerOptions options;
+  options.mutable_base_options()->mutable_model_file()->set_file_name(
+    JoinPath("./" /*test src dir*/, kTestDataDirectory,
+              kESRGANModelWithInputAndOutputMetaData));
+  SUPPORT_ASSERT_OK_AND_ASSIGN(std::unique_ptr<ImageTransformer> image_transformer,
+                      ImageTransformer::CreateFromOptions(options));
+
+  StatusOr<FrameBuffer> result_or =
+      image_transformer->Transform(*husky_downsampled_buffer);
+  SUPPORT_ASSERT_OK(result_or);
+  EXPECT_DOUBLE_EQ(PSNR(result_or.value(), *husky_original_buffer), 25.073790631326489);
```

Comment:
Building on your idea of storing the "husky_test.png" (the output I got from dumping into file => reading in ipynb => saving as png).  I thought of using that "golden image" to compare with `result_or.value()`. The problem is, once again, `LoadImage` due to which the pixels are a bit off. So I think normal PSNR solution is the best stand-in solution atm.